routes/v0/cases.ts

```diff
@@ -1,55 +1,73 @@
+import createError from 'http-errors'
```

Comment:
As well as moving over to the pg-promise API, I moved some of the error handling that produces HTTP errors up into this file. 

In services built with the MVC pattern it's generally good practice to throw generic errors from the Model and keep all the HTTP specific stuff in the Controller (in terms of the MVC pattern, the 'routes' in HRM are acting as the Controller layer, whereas the modules currently called 'controllers' would be considered part of the Model :-/ ), then the model code isn't HTTP specific.

Reply:
I'd say generating HTTP errors in middlewares is totally fine, middlewares are very much part of the 'HTTP domain', so generating HTTP errors in them is entirely appropriate. I just don't want to be generating HTTP errors in the business logic or data access layers, which shouldn't care about the type of application or service they are being hosted in